### PR TITLE
Fix new-rg templates

### LINKED ISSUE #2631: Test for and mitigate "InvalidTemplate" errors in Samples
Beginning June 17th, virtually all ARM templates of the "new resource group" variety used in bot deployments have generated "InvalidTemplate - Deployment template validation failed" errors. They are typically named "template-with-new-rg.json". 

The templates in the BotBuilder-Samples repo should be tested for that error. If they error, they present a pitfall for users attempting to deploy bots to Azure. Monica Rivera is currently pursuing the Azure DevOps team for a solution.

One way to mitigate this problem is to remove the templates from the repo until the problem is solved. Users would still have the "existing resource group" variety of templates to use for bot deployments. Those are typically named "template-with-preexisting-rg.json".

### PR DESCRIPTION
**Pull request by v-kydela**
Fixes #2631

## Proposed Changes

  - "dependsOn" arrays have been modified so that the correct resources can be identified in the inner contexts
  - Indentation has been corrected for consistency
  - The LinuxDotNet templates have been renamed from template-with-new-rg.json to just template.json because they don't create a new resource group and because we don't want customers trying to use them by accident

## Testing

I have confirmed that the templates now deploy successfully using the instructions here: https://aka.ms/azuredeployment

### CODE REVIEW

File: samples/csharp_dotnetcore/25.message-reaction/DeploymentTemplates/template-with-new-rg.json [hunk 1]
```diff
@@ -151,42 +151,44 @@
...
-                        "resources": [
-                          {
-                            "name": "MsTeamsChannel",
-                            "type": "channels",
+                        {
+                            "apiVersion": "2017-12-01",
+                            "type": "Microsoft.BotService/botServices",
+                            "name": "[parameters('botId')]",
                             "location": "global",
-                            "apiVersion": "2018-07-12",
```
Comment (BruceHaley) [inline samples/csharp_dotnetcore/25.message-reaction/DeploymentTemplates/template-with-new-rg.json:178]: Did you notice that here you switched from a newer apiVersion to an older one?

File: samples/csharp_dotnetcore/55.teams-link-unfurling/DeploymentTemplates/template-with-new-rg.json [hunk 2]
```diff
@@ -151,42 +151,44 @@
...
-                        "resources": [
-                          {
-                            "name": "MsTeamsChannel",
-                            "type": "channels",
+                        {
+                            "apiVersion": "2017-12-01",
+                            "type": "Microsoft.BotService/botServices",
+                            "name": "[parameters('botId')]",
                             "location": "global",
-                            "apiVersion": "2018-07-12",
```
Comment (BruceHaley) [inline samples/csharp_dotnetcore/55.teams-link-unfurling/DeploymentTemplates/template-with-new-rg.json:178]: Same here: switched from newer to older apiVersion.

File: samples/csharp_dotnetcore/57.teams-conversation-bot/DeploymentTemplates/template-with-new-rg.json [hunk 3]
```diff
@@ -151,42 +151,44 @@
...
-                        "resources": [
-                          {
-                            "name": "MsTeamsChannel",
-                            "type": "channels",
+                        {
+                            "apiVersion": "2017-12-01",
+                            "type": "Microsoft.BotService/botServices",
+                            "name": "[parameters('botId')]",
                             "location": "global",
-                            "apiVersion": "2018-07-12",
```
Comment (BruceHaley) [inline samples/csharp_dotnetcore/57.teams-conversation-bot/DeploymentTemplates/template-with-new-rg.json:178]: Ditto

File: samples/csharp_dotnetcore/58.teams-start-new-thread-in-channel/DeploymentTemplates/template-with-new-rg.json [hunk 4]
```diff
@@ -151,42 +151,44 @@
...
-                        "resources": [
-                          {
-                            "name": "MsTeamsChannel",
-                            "type": "channels",
+                        {
+                            "apiVersion": "2017-12-01",
+                            "type": "Microsoft.BotService/botServices",
+                            "name": "[parameters('botId')]",
                             "location": "global",
-                            "apiVersion": "2018-07-12",
```
Comment (BruceHaley) [inline samples/csharp_dotnetcore/58.teams-start-new-thread-in-channel/DeploymentTemplates/template-with-new-rg.json:178]: Etcetera

File: samples/csharp_dotnetcore/58.teams-start-new-thread-in-channel/DeploymentTemplates/template-with-new-rg.json [hunk 5]
```diff
@@ -151,42 +151,44 @@
...
                             "kind": "bot",
+                            "sku": {
+                                "name": "[parameters('botSku')]"
+                            },
                             "properties": {
-                              "channelName": "MsTeamsChannel"
```
Comment (BruceHaley) [inline samples/csharp_dotnetcore/58.teams-start-new-thread-in-channel/DeploymentTemplates/template-with-new-rg.json:181]: Apparently this channelName property is not needed?

File: samples/javascript_nodejs/25.message-reaction/deploymentTemplates/template-with-new-rg.json [hunk 6]
```diff
@@ -151,42 +151,44 @@
...
-                        "resources": [
-                          {
-                            "name": "MsTeamsChannel",
-                            "type": "channels",
+                        {
+                            "apiVersion": "2017-12-01",
+                            "type": "Microsoft.BotService/botServices",
+                            "name": "[parameters('botId')]",
                             "location": "global",
-                            "apiVersion": "2018-07-12",
```
Comment (BruceHaley) [inline samples/javascript_nodejs/25.message-reaction/deploymentTemplates/template-with-new-rg.json:178]: Reverting to older apiVersion?

File: samples/javascript_nodejs/25.message-reaction/deploymentTemplates/template-with-new-rg.json [hunk 7]
```diff
@@ -151,42 +151,44 @@
...
                             "kind": "bot",
+                            "sku": {
+                                "name": "[parameters('botSku')]"
+                            },
                             "properties": {
-                              "channelName": "MsTeamsChannel"
```
Comment (BruceHaley) [inline samples/javascript_nodejs/25.message-reaction/deploymentTemplates/template-with-new-rg.json:181]: Dropping channelName property?

File: samples/javascript_nodejs/50.teams-messaging-extensions-search/deploymentTemplates/template-with-new-rg.json [hunk 8]
```diff
@@ -151,42 +151,44 @@
...
-                        "resources": [
-                          {
-                            "name": "MsTeamsChannel",
-                            "type": "channels",
+                        {
+                            "apiVersion": "2017-12-01",
+                            "type": "Microsoft.BotService/botServices",
+                            "name": "[parameters('botId')]",
                             "location": "global",
-                            "apiVersion": "2018-07-12",
```
Comment (BruceHaley) [inline samples/javascript_nodejs/50.teams-messaging-extensions-search/deploymentTemplates/template-with-new-rg.json:178]: Nwerer ==> older?

File: samples/javascript_nodejs/51.teams-messaging-extensions-action/deploymentTemplates/template-with-new-rg.json [hunk 9]
```diff
@@ -151,42 +151,44 @@
...
-                        "resources": [
-                          {
-                            "name": "MsTeamsChannel",
-                            "type": "channels",
+                        {
+                            "apiVersion": "2017-12-01",
+                            "type": "Microsoft.BotService/botServices",
+                            "name": "[parameters('botId')]",
                             "location": "global",
-                            "apiVersion": "2018-07-12",
```
Comment (BruceHaley) [inline samples/javascript_nodejs/51.teams-messaging-extensions-action/deploymentTemplates/template-with-new-rg.json:178]: ==> older?

File: samples/javascript_nodejs/51.teams-messaging-extensions-action/deploymentTemplates/template-with-new-rg.json [hunk 10]
```diff
@@ -151,42 +151,44 @@
...
                             "kind": "bot",
+                            "sku": {
+                                "name": "[parameters('botSku')]"
+                            },
                             "properties": {
-                              "channelName": "MsTeamsChannel"
```
Comment (BruceHaley) [inline samples/javascript_nodejs/51.teams-messaging-extensions-action/deploymentTemplates/template-with-new-rg.json:181]: channelName?

**Review by BruceHaley — CHANGES_REQUESTED** [review]
Your two template renames break these two builds: 
Sample-DotNet-EchoBot-Linux-Test https://fuselabs.visualstudio.com/SDK_v4/_build?definitionId=999 
Sample-DotNet-CoreBot-Linux-Test https://fuselabs.visualstudio.com/SDK_v4/_build?definitionId=1000

**Comment by BruceHaley** [comment]
These new-rg ARM templates are not tested in any of our functional test builds. I suggest they be tested there. Issue #2728

File: samples/csharp_dotnetcore/25.message-reaction/DeploymentTemplates/template-with-new-rg.json (hunk 1 above)
Comment (v-kydela) [reply samples/csharp_dotnetcore/25.message-reaction/DeploymentTemplates/template-with-new-rg.json:178]: That's incorrect. The lines may look a bit jumbled in the diff, but 2017-12-01 is for the bot service and 2018-07-12 is for the Teams channel.

File: samples/csharp_dotnetcore/58.teams-start-new-thread-in-channel/DeploymentTemplates/template-with-new-rg.json (hunk 5 above)
Comment (v-kydela) [reply samples/csharp_dotnetcore/58.teams-start-new-thread-in-channel/DeploymentTemplates/template-with-new-rg.json:181]: This is the red line 181. You'll need to scroll down a bit to find the green line 181. The property remains, but the indentation has been corrected.

**Comment by v-kydela** [comment]
> Your two template renames break these two builds:
> Sample-DotNet-EchoBot-Linux-Test https://fuselabs.visualstudio.com/SDK_v4/_build?definitionId=999
> Sample-DotNet-CoreBot-Linux-Test https://fuselabs.visualstudio.com/SDK_v4/_build?definitionId=1000

Thanks @BruceHaley, can you fix those builds please?

**Comment by BruceHaley** [comment]
> > Your two template renames break these two builds:
> > Sample-DotNet-EchoBot-Linux-Test https://fuselabs.visualstudio.com/SDK_v4/_build?definitionId=999
> > Sample-DotNet-CoreBot-Linux-Test https://fuselabs.visualstudio.com/SDK_v4/_build?definitionId=1000
> 
> Thanks @BruceHaley, can you fix those builds please?

Let me know when you merge so I can coordinate the build fixes. Thanks.

**Review by BruceHaley — APPROVED** [review]

**Comment by v-kydela** [comment]
> Let me know when you merge so I can coordinate the build fixes. Thanks.

I'm not authorized to merge the pull request, and a code owner review is required.

**Comment by v-kydela** [comment]
@microsoft/bb-samples 
@sgellock 
@johnataylor 
@garypretty 
@axelsrz 
@stevengum 
@vishwacsena 

Please merge

**Pull request merged**